### PR TITLE
Revamp modals 1

### PR DESCRIPTION
**Pull request by piensaenpixel**

### CODE REVIEW

**Comment by xavijam** [comment]
50 files changed! 💃

**Comment by xavijam** [comment]
Tests failing are related with basemap dialog, please @matallo, could you check it with @piensaenpixel? 💋

**Comment by josecruz** [comment]
hey guys, I share here some things that I noticed: 

1. "Delete Dataset" modal is still with the old style
image

2. "Export Dataset" modal on the *Dataset View* still with the old style:
image

I'll be updating this.

File: lib/assets/javascripts/cartodb3/components/modals/add-basemap/xyz/xyz.tpl [hunk 1]
```diff
@@ -1,35 +1,15 @@
...
-    <div class="CreateDialog-footerLine"></div>
-    <div class="CreateDialog-footerInner ">
-      <div class="CreateDialog-footerInfo"></div>
-      <div class="CreateDialog-footerActions">
-        <%/* ok class == let parent dialog view handle the click event */%>
-        <button class="CDB-Button CDB-Button--primary is-disabled js-ok ok">
-          <span class="CDB-Button-Text CDB-Text is-semibold CDB-Size-medium u-upperCase"><%- _t('components.modals.add-basemap.add-btn') %></span>
-        </button>
+        <button class="Checkbox-input u-rSpace--m"></button>
+        <label class="CDB-Text CDB-Size-small is-semibold u-upperCase">TMS</label>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/add-basemap/xyz/xyz.tpl]: TMS should be translated.

File: lib/assets/javascripts/cartodb3/components/modals/export-data/modal-export-data-view.js [hunk 2]
```diff
@@ -174,7 +166,6 @@ module.exports = CoreView.extend({
       this.submit();
     }
 
-    this.$('.js-db').attr('disabled', 'disabled');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/export-data/modal-export-data-view.js]: Why have we decided to remove that?

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/export-data/modal-export-data-view.js]: Ah, not appearing in the template... curious.

File: lib/assets/javascripts/cartodb3/data/layer-definition-model.js [hunk 3]
```diff
@@ -255,6 +255,34 @@ module.exports = Backbone.Model.extend({
         });
       }, this)
       .value();
+  },
+
+  getAllDependentLayers: function () {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/layer-definition-model.js]: In my opinion, this method deserves some tests cc @MariaCheca :kiss:.

File: lib/assets/javascripts/cartodb3/editor/layers/delete-layer-confirmation.tpl [hunk 4]
```diff
@@ -1,23 +1,49 @@
...
+        <% if (affectedItemsMessages.length > 0) { %>
+          <li class="Modal-listTextItem">
+            <p class="CDB-Text CDB-Size-medium">
+              <%= _t('editor.layers.delete.affected-items') %>
+              <% for (var i = 0; i < affectedItemsMessages.length; i++) { %>
+                  <span class="CDB-Text is-semibold"><%- affectedItemsMessages[i] %></span>
+                  <% if (i < (affectedItemsMessages.length - 2)) { %>
+                    ,
+                  <% } else if (i == (affectedItemsMessages.length - 2)) { %>
+                    and
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/delete-layer-confirmation.tpl]: `and` should be translated. cc @MariaCheca

**Comment by xavijam** [comment]
screen shot 2016-09-14 at 17 52 07

found this small error @piensaenpixel

**Comment by piensaenpixel** [comment]
Fixed translations and ui issues. Only pending some new tests @MariaCheca  💋

File: lib/assets/javascripts/cartodb3/data/layer-definition-model.js (hunk 3 above)
Comment (MariaCheca) [inline lib/assets/javascripts/cartodb3/data/layer-definition-model.js]: ASAP 👍

**Comment by MariaCheca** [comment]
@xavijam tests added, look if everything is good to you and if yes 🚢 🇮🇹

File: lib/assets/test/jasmine/cartodb3/data/layer-definition-model.spec.js [hunk 5]
```diff
@@ -455,4 +455,68 @@ describe('cartodb3/data/layer-definition-model', function () {
...
+      }, {
+        id: 'e1',
+        type: 'buffer',
+        radio: 300,
+        distance: 'meters',
+        source: 'a0'
+      }]);
+    });
+
+    it('should return the number of layers depending on it', function () {
```
Comment (xavijam) [inline lib/assets/test/jasmine/cartodb3/data/layer-definition-model.spec.js]: Hey! great, but I'd more use cases, not just only one, come on! 💪

**Comment by xavijam** [comment]
Great :es:

**Comment by matallo** [comment]
my bad for not reviewing this properly but right now I'm struggling with the basemaps modals (besides adapting them for wms, where there are several views), there are some cases like this currently happening in production, I'll extend the tests to cover them

c5369f99-7a43-4199-aa76-c17fe48462c7-1

/cc @piensaenpixel @xavijam

**Pull request merged**